### PR TITLE
Snap updates for 3.1 GA

### PR DESCRIPTION
**Pull request by vivmishra**

### CODE REVIEW

File: src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml
Comment (jkotas) [inline src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: Why are we keeping the snap files for older / already shipped versions in master?

Comment (vivmishra) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: @dagood , what would you recommend we do with these snap files as we moved them from core-setup to runtime as per your suggestion? 

Ideally, we would want to keep all of the snap files in one place (one storage location for all) to not litter them across branches as per product release.

Comment (dagood) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: These were put in Core-Setup master a while back just to have a place to check them in, I don't think they have a good home. Ideally they would live near the artifacts they actually use, Core-Setup 3.0/3.1 and Runtime master for the ones that ship the runtime, plus I think Core-SDK 3.0/3.1 and SDK master for the one shipping SDKs. They would also ideally actually be used during the build and produce the snaps as build artifacts. https://github.com/dotnet/core-setup/issues/4789 tracks the overall effort.

The suggestion to move the PR to dotnet/runtime isn't because it belongs here, it's solely because core-setup is now consolidated into this repo. (Core-Setup no longer has a master branch.)

Comment (dagood) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: Context, the Core-Setup PR: https://github.com/dotnet/core-setup/pull/8912

Comment (jkotas) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: The snap files look very similar to docker files. Would it make sense to keep the snap and docker files together?

Comment (vivmishra) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: As long as it is a consistent home (that is not spread across repos) for these snap files it should be fine to keep them in any logical repo as they are independent of anything else. 
Yes, they do look similar to docker files but I am not sure if they fit right in the repo that is specifically created for docker: https://github.com/dotnet/**dotnet-docker**  

Cc: @MichaelSimons

File: src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml
Comment (am11) [inline src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: I think libunwind8 and few other dependencies are not required for 3.0 onwards? cc @janvorli

Comment (janvorli) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: Correct, libunwind8 need was dropped, actually in 2.1 already.
The lldb isn't needed from 3.0 on, since the libsosplugin is not part of the runtime / sdk anymore.
As for the libssl, we support running with libssl1.0.0, 1.1.0 and 1.1.1. On some distros, the 1.0.0 is not available anymore and only 1.1.0/1.1.1 are present. I am not familiar with how snap works though, so I am not sure if that's a problem or not.

Comment (MichaelSimons) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: This would also apply to the SDK image.

File: src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml
Comment (MichaelSimons) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: Adding the snap artifacts to the docker repo would go against the source code repo patterns used within the Docker ecosystem.  The Docker practice is to keep all Dockerfiles in a standalone github repo.  I would not be in favor of putting the snap source in the Docker repo.

File: src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml
Comment (vivmishra) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: I can make the suggested changes, but do we have approval to merge this in runtime repo?

**Review by dagood — APPROVED** [review]
We need a better place for this, but blocking this PR until we find it seems counterproductive. Priority should be having this up to date and available for public eyes.

File: src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml
Comment (vivmishra) [reply src/installer/pkg/packaging/snaps/dotnet-runtime-3.1/snap/snapcraft.yaml]: Fixed

**Pull request merged**
